### PR TITLE
Add front end unit tests Pt.1

### PR DESCRIPTION
**Pull request by dhaselhan**

### CODE REVIEW

File: alcs-frontend/src/app/features/board/board.component.spec.ts [hunk 1]
```diff
@@ -25,4 +160,228 @@ describe('BoardComponent', () => {
...
+        code: 'boardCode',
+        title: 'boardTitle',
+        isFavourite: false,
+        decisionMaker: '',
+        statuses: [],
+      },
+    ]);
+
+    fixture.detectChanges();
+    await sleep(1);
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/board/board.component.spec.ts:220]: 🛌

File: alcs-frontend/src/app/features/board/board.component.spec.ts [hunk 2]
```diff
@@ -25,4 +160,228 @@ describe('BoardComponent', () => {
...
+      {
+        code: 'boardCode',
+        title: 'boardTitle',
+        isFavourite: false,
+        decisionMaker: '',
+        statuses: [],
+      },
+    ]);
+
+    fixture.detectChanges();
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/board/board.component.spec.ts:245]: I think it is worth wrapping this into a helper function with some description regarding sleep. This way once there will be nicer solution we will be able to apply it there

File: alcs-frontend/src/app/services/application/application-decision/application-decision.service.spec.ts [hunk 3]
```diff
@@ -1,19 +1,193 @@
...
+    httpClient.patch.mockReturnValue(
+      throwError(() => {
+        new Error('');
+      })
+    );
+
+    try {
+      await service.update('1', {});
+    } catch (e) {
+      //OM NOM NOM
```
Comment (mhuseinov) [inline alcs-frontend/src/app/services/application/application-decision/application-decision.service.spec.ts:88]: 🍔

File: alcs-frontend/src/app/services/application/application-modification/application-modification.service.ts [hunk 4]
```diff
@@ -58,16 +46,17 @@ export class ApplicationModificationService {
 
   async update(uuid: string, updateDto: ApplicationModificationUpdateDto) {
     try {
-      await firstValueFrom(this.http.patch<ApplicationModificationDto>(`${this.url}/${uuid}`, updateDto));
+      return await firstValueFrom(this.http.patch<ApplicationModificationDto>(`${this.url}/${uuid}`, updateDto));
```
Comment (mhuseinov) [inline alcs-frontend/src/app/services/application/application-modification/application-modification.service.ts:49]: I do not think you need await here

File: alcs-frontend/src/app/services/application/application-reconsideration/application-reconsideration.service.ts [hunk 5]
```diff
@@ -60,18 +47,19 @@ export class ApplicationReconsiderationService {
...
     try {
-      await firstValueFrom(
+      return await firstValueFrom(
         this.http.patch<ApplicationReconsiderationDto>(`${this.url}/${reconsiderationUuid}`, reconsideration)
       );
     } catch (err) {
       console.error(err);
       this.toastService.showErrorToast('Failed to update reconsideration');
     }
+    return undefined;
```
Comment (mhuseinov) [inline alcs-frontend/src/app/services/application/application-reconsideration/application-reconsideration.service.ts:57]: Just 'return;' for consistency with the rest of the code?

**Review by mhuseinov — APPROVED** [review]

File: alcs-frontend/src/app/features/board/board.component.spec.ts (hunk 2 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/features/board/board.component.spec.ts]: Removed fixture.detectChanges as its not needed

File: alcs-frontend/src/app/services/application/application-reconsideration/application-reconsideration.service.ts (hunk 5 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/services/application/application-reconsideration/application-reconsideration.service.ts]: Done!

**Pull request merged**